pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -435,13 +435,7 @@ func (fs *Decomposedfs) ListStorageSpaces(ctx context.Context, filter []*provide
 // MustCheckNodePermissions checks if permission checks are needed to be performed when user requests spaces
 func (fs *Decomposedfs) MustCheckNodePermissions(ctx context.Context, requestedUserID string, unrestricted bool) bool {
 	canListAllSpaces := fs.canListAllSpaces(ctx)
-	switch {
-	case canListAllSpaces && !unrestricted:
-		return true
-	case (canListAllSpaces && requestedUserID == userIDAny):
-		// admins should not see any other spaces when they request their own, without settings filters
-		return true
-	case canListAllSpaces, unrestricted:
+	if canListAllSpaces && unrestricted {
```

Comment:
Is this supposed to be a simplification or a logic change? It seems to be slightly different, e.g. `!canListAllSpaces` / `unrestricted` will return `true` now but returned `false` before.

Reply:
That is a change. IMO that is the correct behavior. The case `!canListAllSpaces` / `unrestricted` should return `true``

`!canListAllSpaces` means the user has not the permission from the role and we need to check the grants.

Reply:
This is the second time I am reading a PR on this and I still haven't understood what `unrestricted` means. Could you please add a comment so I understand next time?

Because from my naive understanding: If something is unrestricted I should not need to check node permissions for it.

Reply:
will do. `unrestricted` means the user is trying to list all drives (calling `/drives` instead of `/me/drives`), but this should only be possible if we have `canListAllSpaces`

Reply:
done